### PR TITLE
Enhance checkbox

### LINKED ISSUE #36: [Syncing] Checkbox
## Points to note
- Component in action - `Checkbox.js`
- Github Urls to this component:
  * [neeto-ui/Checkbox.js](https://github.com/bigbinary/neeto-ui/tree/67ee4a5188c5feccbeca2767107651fb24932764/lib/components/Checkbox.js)
  * [NitroUI/Checkbox.js](https://github.com/bigbinary/neeto-help-web/tree/0b11a7b0f98ac7c533df1e9ec37cf900b4cbc901/app/javascript/Admin/Components/NitroUI/Checkbox.js)
- :green_circle: Green color signifies how that particular line is written in `neeto-ui` package
- :red_circle: Red color signifies how that particular line is written in `NitroUI` internals

## Sync Required or Not
- Are there significant changes for `Checkbox.js` that needs to be synced to `neeto-ui` package from `NitroUI`? - **No**

## Diff
```diff
--- NitroUI/Checkbox.js	2020-11-28 16:54:23.861319500 +0530
+++ neetoui-official/Checkbox.js	2020-11-28 16:49:18.735640100 +0530
@@ -14,30 +14,31 @@
   const uniqueId = uuid();
 
   return (
-    <div className="relative flex flex-row items-center justify-start">
-      <div className="flex flex-row items-center justify-center h-5">
-        <input
-          name={name}
-          type="checkbox"
-          className={classnames([
-            "w-4 h-4 text-purple-500 transition duration-200 ease-in-out form-checkbox cursor-pointer",
-            className
-          ])}
-          data-test-id={dataTestId}
-          id={id || uniqueId}
-          onChange={() => {}}
-          {...otherProps}
-        />
+    <>
+      <div className="relative flex items-start">
+        <div className="flex items-center h-5">
+          <input
+            name={name}
+            type="checkbox"
+            className={classnames([
+              "w-4 h-4 text-purple-500 transition duration-200 ease-in-out form-checkbox cursor-pointer",
+              className
+            ])}
+            data-test-id={dataTestId}
+            id={id || uniqueId}
+            {...otherProps}
+          />
+        </div>
+        {(label || required) && (
+          <label
+            htmlFor={id || uniqueId}
+            className="mb-0 ml-2 text-sm font-medium leading-5 text-gray-700"
+          >
+            {label || (required ? <span className="text-red-700">*</span> : "")}
+          </label>
+        )}
       </div>
-      {(label || required) && (
-        <label
-          htmlFor={id || uniqueId}
-          className="mb-0 ml-2 text-sm font-medium leading-5 text-gray-700"
-        >
-          {label || (required ? <span className="text-red-700">*</span> : "")}
-        </label>
-      )}
-    </div>
+    </>
   );
 };
 
```

### PR DESCRIPTION
**Pull request by yedhink**
Fixes #36 

@goutham-subramanyam _a

Please review. Let me know if we need to make anymore changes.

### CODE REVIEW

File: lib/components/Checkbox.js [hunk 1]
```diff
@@ -1,44 +1,41 @@
...
+          {...otherProps}
+        />
       </div>
-    </>
+      {(label || required) && (
+        <Label
+          htmlFor={id || uniqueId}
+          className="mb-0 ml-2 text-sm font-medium leading-5 text-gray-700"
+        >
+          {label || (required ? <span className="text-red-700">*</span> : "")}
```
Comment (edwinbbu) [inline lib/components/Checkbox.js:35]: @yedhink _a
Please remove red color for required. We are changing it into black in all the places.
```suggestion
          {label || (required ? <span className="ml-0.5">*</span> : "")}
```

Comment (goutham-subramanyam) [reply lib/components/Checkbox.js:35]: required is handled inside the Label component right. So we just need to pass the prop.

File: lib/components/Checkbox.js [hunk 2]
```diff
@@ -1,44 +1,41 @@
...
+          ])}
+          id={id || uniqueId}
+          {...otherProps}
+        />
       </div>
-    </>
+      {(label || required) && (
+        <Label
+          htmlFor={id || uniqueId}
+          className="mb-0 ml-2 text-sm font-medium leading-5 text-gray-700"
```
Comment (goutham-subramanyam) [inline lib/components/Checkbox.js:33]:
```suggestion
          className="ml-3"
```

**Review by goutham-subramanyam — CHANGES_REQUESTED** [review]

**Comment by yedhink** [comment]
@goutham-subramanyam _a

Please review.

**Review by goutham-subramanyam — APPROVED** [review]
LGTM.

**Pull request merged**